status/status.go

```diff
@@ -99,26 +99,26 @@ func FromError(err error) (s *Status, ok bool) {
 	}
 	type grpcstatus interface{ GRPCStatus() *Status }
 	if gs, ok := err.(grpcstatus); ok {
-		grpcStatus := gs.GRPCStatus()
-		if grpcStatus == nil {
+		if gs == nil || gs.GRPCStatus() == nil {
```

Comment:
Sorry to contradict the previous reviewer.  There was some effort taken in the past to avoid calling `gs.GRPCStatus()` multiple times.  Let's try to keep that condition true.  E.g.


```go
type grpcstatus interface { GRPCStatus() *Status }
var gs grpcstatus
if gs, ok := err.(grpcstatus); ok && gs != nil {
	if grpcStatus := gs.GRPCStatus(); grpcStatus != nil {
		return grpcStatus, true
	}
} else if errors.As(err, &gs) && gs != nil {
	if grpcStatus := gs.GRPCStatus(); grpcStatus != nil {
		p := grpcStatus.Proto()
		p.Message = err.Error()
		return status.FromProto(p), true
	}
}
// Error is or wraps an error that has status nil, which maps
// to codes.OK..................
return New(codes.Unknown, err.Error()), false
```

Reply:
Thanks @dfawley for letting me know about the constraint for `GRPCStatus` , I will update the PR as per your suggestions